python_modules/dagster/dagster/core/definitions/pipeline.py

```diff
@@ -613,7 +614,11 @@ def _get_pipeline_subset_def(
                 ),
             )
 
-    solids = list(map(graph.solid_named, solids_to_execute))
+    # go in topo order to ensure deps dict is ordered
+    solids = list(
+        filter(lambda solid: solid.name in solids_to_execute, graph.solids_in_topological_order)
```

Comment:
why was this causing a grpc error?

Reply:
it was throwing a check exception in the user code when building the subset pipeline since the dynamic output resolution alogrithm expects to process the dependencies in order